lib/workers/pr/changelog/source-gitlab.ts

```diff
@@ -0,0 +1,159 @@
+import URL from 'url';
+import { Release } from '../../../datasource';
+import { logger } from '../../../logger';
+import { api } from '../../../platform/gitlab/gl-got-wrapper';
+import * as globalCache from '../../../util/cache/global';
+import { regEx } from '../../../util/regex';
+import * as allVersioning from '../../../versioning';
+import { BranchUpgradeConfig } from '../../common';
+import { ChangeLogRelease, ChangeLogResult } from './common';
+import { addReleaseNotes } from './release-notes';
+
+const { get: glGot } = api;
+
+const cacheNamespace = 'changelog-gitlab-release';
+
+async function getTags(
+  endpoint: string,
+  versionScheme: string,
+  repository: string
+): Promise<string[]> {
+  logger.trace('getTags() from gitlab');
+  let url = endpoint
+    ? endpoint.replace(/\/?$/, '/')
+    : /* istanbul ignore next: not possible to test, maybe never possible? */ 'https://gitlab.com/api/v4/';
```

Comment:
```suggestion
  let url = endpoint;
```
`endpoint` is always passed with ending slash now